javascript-modules-engine/pom.xml

```diff
@@ -268,7 +269,7 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.11.0</version>
                 <configuration>
-                    <release>11</release>
+                    <release>17</release>
```

Comment:
This should also be changed for the `javascript-modules-engine-java` module and very likely in the Github actions as well, no?
On that note, I'm wondering if we should support JDK 17 at build time (meaning developers should install and configure a JDK 17+ for development) or stick with JDK 11 as the rest of the modules, and only make the JDK 17 a requirement at runtime, what do you think?